policy/modules/services/container.te

```diff
@@ -0,0 +1,880 @@
+allow container_engine_domain self:capability { chown dac_override dac_read_search fowner fsetid kill mknod net_admin net_raw setfcap setpcap setgid setuid sys_admin sys_chroot sys_ptrace sys_resource };
+allow container_engine_domain self:capability2 { bpf perfmon };
+allow container_engine_domain self:bpf { map_create map_read map_write prog_load prog_run };
+allow container_engine_domain self:cap_userns { audit_write chown dac_override dac_read_search fowner fsetid ipc_lock ipc_owner kill lease linux_immutable mknod net_admin net_bind_service net_raw setfcap setgid setpcap setuid sys_admin sys_boot sys_chroot sys_nice sys_pacct sys_ptrace sys_rawio sys_resource sys_time sys_tty_config };
+allow container_engine_domain self:cap2_userns { audit_read bpf block_suspend perfmon syslog wake_alarm };
+allow container_engine_domain self:bpf { map_create map_read map_write prog_load prog_run };
+allow container_engine_domain self:fd use;
+allow container_engine_domain self:netlink_route_socket create_netlink_socket_perms;
+allow container_engine_domain self:packet_socket create_socket_perms;
+allow container_engine_domain self:fifo_file manage_fifo_file_perms;
+allow container_engine_domain self:tcp_socket create_stream_socket_perms;
+allow container_engine_domain self:unix_stream_socket { create_stream_socket_perms connectto };
+allow container_engine_domain self:unix_dgram_socket { create_socket_perms sendto };
+
+allow container_engine_domain container_port_t:tcp_socket name_bind;
+
+dontaudit container_engine_domain container_domain:process { noatsecure rlimitinh siginh };
+allow container_engine_domain container_domain:process2 { nnp_transition nosuid_transition };
+
+allow container_engine_domain container_mountpoint_type:dir_file_class_set mounton;
+
+corecmd_bin_entry_type(container_engine_domain)
+corecmd_exec_bin(container_engine_domain)
+# needed when spawning interactive shells inside containers
+corecmd_exec_shell(container_engine_domain)
+corecmd_search_bin(container_engine_domain)
+# podman unshare causes most of this noise
+corecmd_dontaudit_exec_all_executables(container_engine_domain)
+
+corenet_tcp_bind_generic_node(container_engine_domain)
+corenet_tcp_connect_http_port(container_engine_domain)
+corenet_tcp_bind_all_ports(container_engine_domain)
+corenet_udp_bind_all_ports(container_engine_domain)
+corenet_rw_tun_tap_dev(container_engine_domain)
+
+dev_getattr_all_blk_files(container_engine_domain)
+dev_getattr_all_chr_files(container_engine_domain)
+dev_getattr_fs(container_engine_domain)
+dev_remount_fs(container_engine_domain)
+dev_list_sysfs(container_engine_domain)
+# required by crun
+dev_read_sysfs(container_engine_domain)
+dev_mount_sysfs(container_engine_domain)
+dev_remount_sysfs(container_engine_domain)
+dev_mounton_sysfs_dirs(container_engine_domain)
```

Comment:
Why isn't there a crun domain?

Reply:
If executed directly `crun` will run in `container_engine_t`. Since podman and docker could be configured to use either `crun` or `runc` I didn't think it was necessary to create new domains for these low-level container runtimes. However, I didn't add a `container_engine_user_t` for regular users to run these and realize now I probably should.

I also want to point out that the comment there stating `# required by crun` only applies to `dev_read_sysfs()`.

Reply:
Then the command and rule should be at the bottom of the block, because I can't tell otherwise.